src/k2/tso/client/tso_clientlib.h

```diff
@@ -73,6 +71,10 @@ class TSO_ClientLib
 
     bool _stopped{false};
 
+    // a promise/signal for ready to serve request when they come earlier than TSO server end point set up
+    volatile bool _readyToServe {false};

```

Comment:
No need for volatile since it is still single-threaded

Reply:
Maybe a little bit paranoid here, but the  readyPromise.set_value(); could be executed before this line otherwise and could cause this task be schedule out while another client request may come in and wrong adding another promise and wait on it.

Reply:
removed unnecessary volatile.